emloop/cli/common.py

```diff
@@ -213,7 +213,25 @@ def create_hooks(config: dict, model: AbstractModel,
     return hooks
 
 
-def run(config: dict, output_root: str, restore_from: str=None, eval: Optional[str]=None) -> None:
+def create_main_loop(config: dict, model: Optional[AbstractModel]=None, dataset: Optional[AbstractDataset]=None,
+                     hooks: Optional[List[AbstractHook]]=None):
+    """
+    Create a main loop object.
+
+    :param config: config dict with main loop config
+    :param dataset: dataset object implementing the :py:class:`emloop.datasets.AbstractDataset` concept
+    :param model: model object implementing the :py:class:`emloop.models.AbstractModel` concept
+    :param hooks: list of hooks implementing the :py:class:`emloop.models.AbstractHook` concept
+
+    :return: main loop object
+    """
+
+    logging.info('Creating main loop')
+    main_loop_kwargs = copy.deepcopy(config.get('main_loop', {}))
+    return MainLoop(model=model, dataset=dataset, hooks=hooks, **main_loop_kwargs)
+
+
+def run(config: dict, output_root: str, restore_from: str=None, eval_stream: Optional[str]=None) -> None:
```

Comment:
Do we need to modify (and expose) this method? Basically it does something similar to *create_all* and then calls a single main loop's method. I would rather force the user to create the main loop and call the method for himself instead of *hacking* the config...

Reply:
I did not modify this function, I only renamed one of its arguments as Filip told me to do in his review. Also this function is not exposed in the API.

Reply:
Well ,there are some changes as you create the main loop via `create_main_loop`, perhaps the old direct way was better. I do not like the conditional config dict updates.

Reply:
Yeah, we discussed the conditional update with beda and we dont like it either.

Reply:
I think the current version can be merged. If you don't like it for some reason, please suggest an alternative.

Reply:
Nah, we need to change this. Will think about it tommorrow.